MantidQt/CustomInterfaces/inc/MantidQtCustomInterfaces/EnggDiffraction/EnggDiffractionViewQtGUI.h

```diff
@@ -257,15 +257,15 @@ private slots:
   Ui::EnggDiffractionQtTabFocus m_uiTabFocus;
   Ui::EnggDiffractionQtTabPreproc m_uiTabPreproc;
   // Ui::EnggDiffractionQtTabFitting m_uiTabFitting;
-  EnggDiffFittingViewQtWidget *m_fittingWidget;
+  EnggDiffFittingViewQtWidget *m_fittingWidget{nullptr};
   Ui::EnggDiffractionQtTabSettings m_uiTabSettings;
 
   /// converts QList to a vector
   std::vector<std::string> qListToVector(QStringList list,
                                          bool validator) const;
 
   /// instrument selected (ENGIN-X, etc.)
-  std::string m_currentInst;
+  std::string m_currentInst{"ENGINX"};
```

Comment:
I see, in trying to use in-class initializers as much as possible probably this creates some confusion. And also the `IEnggDiffractionView` shouldn't have been left there.

Could we come up with a general best practice/recommendation from now on? If in cases like this that require a constructor we go for initialization in the constructor init list, but when the constructor is not required we prefer in-class initializers that could also cause some confusion. My assumption is that we want/wish to gradually increase the use of in-class initializers?